packages/web/src/scss/tools/_placement.scss

```diff
@@ -102,31 +118,88 @@ $_placement-inverse-map: (
     );
 }
 
-// Function to return an inverse placement name
-// Example: inverse('top') will return 'bottom'
-// Example: inverse('top-start') will return 'bottom'
-// Example: inverse('top-start', true) will return 'bottom-start'
-// Example: inverse('top-start', true, false) will return 'bottom start'
-@function inverse($placement, $main-axis-only: false, $hyphenate: true) {
+// Function to get the cross-axis direction
+// Example: get-cross-axis-direction('left') will return 'vertical'
+// Example: get-cross-axis-direction('bottom') will return 'horizontal'
+@function get-cross-axis-direction($main-axis) {
+    @if $main-axis == 'left' or $main-axis == 'right' {
+        @return 'vertical';
+    }
+
+    @return 'horizontal';
+}
+
+// Function to check if a placement is logical
+// Example: is-logical('top') will return false
+// Example: is-logical('start') will return true
+@function is-logical($placement) {
+    @return if(string.index($placement, 'start') or string.index($placement, 'end'), true, false);
+}
+
+// Function to return a physical placement name
+// Example: logical-to-physical('start', $direction: 'horizontal') will return 'left'
+@function logical-to-physical($logical, $direction) {
+    @return map.get($_logical-to-physical-placement-map, $direction, $logical);
+}
+
+// Function to transform axis
+// Example: transform-axis('left') will return 'left' (no transformation)
+// Example: transform-axis('left', $inverse: true) will return 'right'
+// Example: transform-axis('start', $inverse: false, $physical-direction: 'horizontal') will return 'left'
+// Example: transform-axis('start', $inverse: true, $physical-direction: 'horizontal') will return 'right'
+@function transform-axis($axis, $inverse: false, $physical-direction: null) {
+    $axis-transformed: $axis;
+
+    @if is-logical($axis-transformed) {
+        @if $inverse {
+            $axis-transformed: map.get($_logical-placement-inverse-map, $axis);
+        }
+
+        @if $physical-direction {
+            $axis-transformed: logical-to-physical($axis-transformed, $physical-direction);
+        }
+    } @else if $inverse {
+        $axis-transformed: map.get($_physical-placement-inverse-map, $axis);
+    }
+
+    @return $axis-transformed;
+}
+
+// Function to transform placement
+// Example: transform('top-start') will return 'top-start' (no transformation)
+// Example: transform('top-start', $main-axis-inverse: true) will return 'bottom-start'
+// Example: transform('top-start', $main-axis-inverse: true, $cross-axis-inverse: true) will return 'bottom-end'
+// Example: transform('top-start', $main-axis-inverse: true, $cross-axis-inverse: true, $cross-axis-physical: true) will return 'bottom-right'
+// Example: transform('top-start', $hyphenate: false) will return 'top start'
+@function transform(
```

Comment:
Doesn't this function do too much transformation? It is ok from my point of view that the function transforms the axis but the hyphenation is a different kind of transformation. 🤔 Shouldn't there be a separate function for that?

Reply:
Or maybe just pass a string for join as an attribute and set the default to the hyphen?

Reply:
I renamed the attribute to join-with.

Reply:
Better, thank you